pkg/core/codeset_client.go

```diff
@@ -0,0 +1,199 @@
+package fuseml
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os/exec"
+	"path"
+	"time"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/pkg/errors"
+
+	codeset "github.com/fuseml/fuseml-core/gen/codeset"
+	giteac "github.com/fuseml/fuseml-core/pkg/core/gitea"
+)
+
+var (
+	// FIXME: generate this and put it in a secret
+	HookSecret = "generatedsecret"
+
+	// StagingEventListenerURL should not exist
+	// FIXME: detect this based on namespaces and services
+	StagingEventListenerURL = "http://el-mlflow-listener.fuseml-workloads:8080"
```

Comment:
Note this fixed value. This is one listener created by fuseml tekton deployment. It might be necessary to always connect this to a new listener that executes specific pipeline...